models/pwa.js

```diff
@@ -31,6 +32,14 @@ const E_MANIFEST_ERROR = exports.E_MANIFEST_ERROR = 2;
 function mergeManifest(pwa, manifest) {
   pwa.name = manifest.name;
   pwa.startUrl = manifest.start_url || '';
+
+  // We need an absolut URL to the PWA (Manifest can include both types)
+  if (manifest.start_url.indexOf('http://') === 0 || manifest.start_url.indexOf('https://') === 0) {
+    pwa.absolutStartUrl = manifest.start_url || '';
+  } else {
+    pwa.absolutStartUrl = uri(manifest.start_url).absoluteTo(manifest.url).toString() || '';
+  }
+
```

Comment:
I don't think you need to check to see if `start_url` is relative or not: I think you can just throw whatever we get from the manifest at `urijs` and let it figure it out:

```js
Uri("/?homescreen=1").absoluteTo("https://example.com/manifest.json").toString()
// -> https://example.com/?homescreen=1
Uri("https://foo.com/something/?homescreen=1").absoluteTo("https://example.com/manifest.json").toString()
// -> https://foo.com/something/?homescreen=1
```